src/server/services/checkWebsiteStatus.js

```diff
@@ -0,0 +1,52 @@
+const cron = require('node-cron');
+const { getConnection } = require('../db');
+const { logMailer } = require('../logMailer');
+const LogEmail = require('../models/LogEmail');
+const colors = require('colors');
+  
+async function checkWebsite() {
+  const conn = getConnection();
+
+  try {
+    const response = await fetch('https://httpstat.us/500/', { method: 'HEAD' });
+
+    if (response.ok) {
+      console.log('The server is up'.green.inverse);
+    } else {
+      console.error('The server is down'.red.inverse);
+
+      const errorMessage = 'The server at https://httpstat.us/500/ is down.';
+
+      // Log the error in the database
+      const log = new LogEmail(undefined, errorMessage);
+      await log.insert(conn);
+
+      // Send a logging email
+      await logMailer(conn);
+    }
+  } 
+  catch (error) {
+    console.error('Error:', error);
```

Comment:
I'm trying to understand what error would cause this. If it was the fetch then maybe this makes sense. On the other hand, if the error occurred while logging then it may happen again when that is tried. Could you help me understand better?

Reply:
I included this catch error to catch non-http errors such as when internet is unavailable, invalid url, and etc. Not sure if this is necessary but I thought I would include it and get some feedback. What do you think?

Reply:
That makes sense. If the code changes to use the standard OED logging then a lot of this will be minimized. Another thought is to limit the try/catch to the fetch so the source of the exception is clearer. A second try/catch could be used for other code but I don't know if that is needed here. If the ``response`` variable is declared before the try/catch then maybe it could be analyzed after the try/catch which you can make only execute if there is no exception to the catch block. What do you think?